sniffles/ruletrafficgenerator.py

```diff
@@ -990,15 +990,30 @@ class BackgroundTraffic(TrafficStream):
     def __init__(self, rule=None, sconf=None, start_sec=-1, start_usec=0):
         super().__init__(None, sconf)
         self.proto = rule.getProto()
-        self.sport = Port(rule.getSport())
-        self.dport = Port(rule.getDport())
+        self.protocolType = rule.getProtocolType()
         self.rule = rule
         self.rand = False
         self.current_seq_a_to_b = random.randint(0, 4000000000)
         self.current_ack_a_to_b = 0
         self.current_seq_b_to_a = random.randint(0, 4000000000)
         self.current_ack_b_to_a = 0
 
+        if self.protocolType == 'http':
+            self.sport = Port('any')
+            self.dport = Port('http')
+        elif self.protocolType == 'ftp':
+            self.sport = Port('ftp')
```

Comment:
The impact of the authenticity of the traffic will depend greatly on what is being evaluated.  If the background is trivially disregarded, then it will have little impact on the device under test.  Conversely, if the traffic exerts a strong impact on the target then it will greatly affect the device under test.  In most cases, close is good enough.  However, well formed traffic is always better than obviously incorrect traffic.  Now, to discuss a little about direction.  Direction is primarily for determining the initiator of a conversation.  Reversing sever to client does not mean that the server is arbitrarily sending data so much that the data is coming from the server side.  This was originally important because Snort makes a distinction between home addresses and foreign addresses.  Thus, the idea of server to client, for Sniffles, really just means the traffic originates from a home address.  Second, there should be valid request traffic for all of the protocols, not just http.  However, you might have to go look at the RFC to find the exact parameters.  However, those codes may not be generic and may need some level of simulation.  Finally, for true background traffic it would be good to randomize as much as possible, in particular the requested URL.  I hope this helps answers some questions.  Please feel free to Slack me if you need more answers.